core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -2146,9 +2146,8 @@ public void notifyNewBestBlock(StoredBlock block) throws VerificationException {
                         // included once again. We could have a separate was-in-chain-and-now-isn't confidence type
                         // but this way is backwards compatible with existing software, and the new state probably
                         // wouldn't mean anything different to just remembering peers anyway.
-                        if (confidence.incrementDepthInBlocks() > context.getEventHorizon())
+                        if (confidence.getDepthInBlocks(getLastBlockSeenHeight()) > context.getEventHorizon())
                             confidence.clearBroadcastBy();
-                        confidenceChanged.put(tx, TransactionConfidence.Listener.ChangeReason.DEPTH);
```

Comment:
@schildbach I suppose this is the line that is causing the so called "useless work" referenced here: https://github.com/bitcoinj/bitcoinj/issues/877 ?